bin/combinetf2_plot_hists_uncertainties.py

```diff
@@ -576,34 +577,35 @@ def main():
         config=config,
     )
 
-    for projection in args.project:
-        channel = projection[0]
-        axes = projection[1:]
-        info = channel_info[channel]
-        if len(axes) == 0:
-            axes = ["yield"]
-        result = fitresult["channels"][channel]["projections"]["_".join(axes)]
+    if len(args.physicsModel) == 0:
+        models = [["basemodel"]]
+    else:
+        models = args.physicsModel
 
-        make_plots(
-            outdir,
-            result,
-            axes=[a for a in info["axes"] if a.name in axes],
-            channel=channel,
-            lumi=info.get("lumi", None),
-            **opts,
-        )
+    for margs in models:
+        model = margs[0]
 
-    for channel, info in channel_info.items():
-        result = fitresult[f"channels"][channel]
+        for channel, info in channel_info.items():
 
-        make_plots(
-            outdir,
-            result,
-            axes=info["axes"],
-            channel=channel,
-            lumi=info.get("lumi", None),
-            **opts,
-        )
+            if model == "basemodel":
```

Comment:
Ideally there would be enough metadata to avoid needing to hardcode behaviour based on which model was used?

Reply:
This is needed only for the basemodel since the basemodel is unlike other models stored on a higher level in the results dictionary, directly under the channel key. The other models are stored under [<model name>][<identifier>] to avoid naming conflicts between different models. We could put basemodel some levels lower under ["basemodel"][<identifier>] for the sake of consistency with other models, then this check would not be needed. But given that the basemodel is a special case I think it's fine to keep it as it is.

Reply:
No, not anymore